### DISCUSSION TITLE
@ant-design/cssinjs 如何和antd版本保持一致？

### DISCUSSION THREAD

**Discussion by zhupp034** [ant-design/ant-design#48139] [General]
### What problem does this feature solve?

目前项目安装了antd和@ant-design/cssinjs，但是很容易两者版本不一致，导致css降级失效。如何保证css降级不会因为每次发版版本问题导致失效？

### What does the proposed API look like?

css降级不会因为哪次pnpm i 之后，导致失效

**Comment by afc163** [comment]
> 目前项目安装了antd和@ant-design/cssinjs，但是很容易两者版本不一致

依赖怎么写的，范围给得够宽应该不会出现不一致。

**Reply by arvinxx** [reply]
最怕的就是锁版本锁错

**Reply by zhupp034** [reply]
<img width="466" alt="image" src="https://github.com/ant-design/ant-design/assets/33925263/6114ce04-06ba-4f5e-8418-e838a5965d50">
之前是这样的，结果前几天你们应该版本升级了，导致线上打包出来的版本不一样了，降级直接没了

**Reply by afc163** [reply]
我知道了，antd 这里依赖不应该更新的太勤：https://github.com/ant-design/ant-design/blob/master/package.json#L119

**Reply by afc163** [reply]
@zhupp034 有锁依赖么，看看 pnpm-lock.yaml 里的依赖怎么写的。

**Reply by arvinxx** [reply]
@afc163 但有可能 antd 是不得不更新的。而这种时候用户的心智可能只会想着升级 antd，不会意识到需要更新 cssinjs

**Reply by Wxh16144** [reply]
看看我前面那个 pr 评论的，自己审查一下 antd 依赖。严格遵守 semver 规范就好了

**Reply by afc163** [reply]
antd 里声明个 `^1.8.0` 就够了，小 patch 一般没有致命问题。

**Reply by zhupp034** [reply]
[pnpm-lock.json](https://github.com/ant-design/ant-design/files/14783668/pnpm-lock.json)
这是之前的lock文件  @afc163

**Comment by Wxh16144** [comment]
```bash
jq '{name, version, dependencies: .dependencies | to_entries | map(select(.key | startswith("@ant-design/"))) | from_entries}' node_modules/antd/package.json
```

![image](https://github.com/ant-design/ant-design/assets/32004925/3aca20f1-d970-4441-a383-73e04586d45d)

**Comment by yoyo837** [comment]
lock文件不要ignore即可

**Comment by afc163** [comment]
See https://github.com/ant-design/ant-design/pull/48141

**Comment by ISyangk** [comment]
我也遇到同样的问题，我的antd版本是5.21.6，@ant-design/cssinjs用的1.22.1，但是降级没有成功，在低版本中控制台未发现antd相关的样式
![image](https://github.com/user-attachments/assets/57100361-76ce-4cff-af45-fcd5879c1991)

![image](https://github.com/user-attachments/assets/44e14136-69d5-4d74-a0e2-c3cb8b9fb031)

![image](https://github.com/user-attachments/assets/413c4002-bcaf-43e0-80df-ab28311e701d)

**Reply by ISyangk** [reply]
@afc163 能帮忙看看什么原因吗？

**Reply by Wxh16144** [reply]
预期是啥？清空 node_modules 和 lock 文件重新安装依赖试试看呗

**Reply by ISyangk** [reply]
预期就是能兼容低版本浏览器，达到样式降级的效果

**Reply by ISyangk** [reply]
解决了，只是有些global里写的antd的样式被覆盖了
